src/tests/infraestructure/controller/TestController.ts

```diff
@@ -0,0 +1,193 @@
+        page: Number(page),
+        limit: Number(limit),
+        search: search as string | undefined,
+        companyId: companyId as string | undefined,
+        difficulty: difficulty as 'EASY' | 'MEDIUM' | 'HARD' | undefined,
+        orderBy: orderBy as string | undefined,
+      });
+
+      res.status(200).json({
+        success: true,
+        ...result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      res.status(500).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async getTestById(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.getById.execute(id);
+      res.status(200).json({
+        success: true,
+        data: result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      const status = errorMessage === 'Test not found' ? 404 : 500;
+      res.status(status).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async startTest(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+      const { userId } = req.body;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      if (!userId) {
+        res.status(400).json({
+          success: false,
+          error: 'userId is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.start.execute(id, userId);
+      res.status(201).json({
+        success: true,
+        data: result,
+      });
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error';
+      const status = errorMessage.includes('not found')
+        ? 404
+        : errorMessage.includes('not active')
+          ? 400
+          : 500;
+      res.status(status).json({
+        success: false,
+        error: errorMessage,
+      });
+    }
+  }
+
+  async getQuestions(req: Request, res: Response): Promise<void> {
+    try {
+      const { id } = req.params;
+      const { sessionId } = req.query;
+
+      if (!id) {
+        res.status(400).json({
+          success: false,
+          error: 'Test ID is required',
+        });
+        return;
+      }
+
+      if (!sessionId || typeof sessionId !== 'string') {
+        res.status(400).json({
+          success: false,
+          error: 'sessionId is required',
+        });
+        return;
+      }
+
+      const result = await services.tests.getQuestions.execute(id, sessionId);
```

Comment:
getQuestions calls services.tests.getQuestions.execute(id, sessionId) without any proof the requester owns the session. The controller does not extract authenticated user information to check ownership.

**Sugerencia:**
```typescript
Require authentication middleware and pass the authenticated user id to the use-case (or check session ownership in controller). Example: await services.tests.getQuestions.execute(id, sessionId, req.user.id). Priority: critical (security)
```